### PR TITLE
CSIT-913 Create test for QATEST-116798 Self Exclusion [Session and login limits]

### PR DESCRIPTION
**Pull request by Alena-deriv**
Test link: https://app.clickup.com/t/20696747/CSIT-913

Test sets Self Exclusion for a client and checks restrictions are applied ([CSIT-1119](https://app.clickup.com/t/20696747/CSIT-1119)):
- Can login back
- Cannot create new real account
- Cannot trade
- Cannot deposit

Test is built on desktop as date picker in mobile view is not accessible, will need to check for workaround.
BO side checks will be covered in a separate [subtask](https://app.clickup.com/t/20696747/CSIT-1120) ( separate PR)

### CODE REVIEW

File: cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  cy.get('[data-duration="13 Days"]').click()
+  cy.findByRole('button', { name: 'Next' }).click()
+  cy.findByRole('button', { name: 'Accept' }).click()
+  cy.findByRole('button', { name: 'Yes, log me out immediately' }).click()
+}
+
+function checkSelfExclusionIsSet() {
+  cy.c_login()
+  cy.c_visitResponsive('/appstore/traders-hub', 'large')
+  cy.wait(2000)
```
Comment (meeyun-deriv) [inline cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js:21]: why do we need different hard wait in this PR?

File: cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  cy.get('input[name="timeout_until"]').click()
+  cy.get('[data-duration="13 Days"]').click()
+  cy.findByRole('button', { name: 'Next' }).click()
+  cy.findByRole('button', { name: 'Accept' }).click()
+  cy.findByRole('button', { name: 'Yes, log me out immediately' }).click()
+}
+
+function checkSelfExclusionIsSet() {
+  cy.c_login()
+  cy.c_visitResponsive('/appstore/traders-hub', 'large')
```
Comment (meeyun-deriv) [inline cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js:20]: can we also include responsive test as we need to prioritise responsive test first

**Review by meeyun-deriv — CHANGES_REQUESTED** [review]
some of the expected behaviour are not included in the current test e.g. after self exclusion, client cant deposit, trader and create new sibling account. 

Please help to add all necessary validations

File: cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js (hunk 1 above)
Comment (Alena-deriv) [reply cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js]: Updated to the common approach - changed to timeouts, to let the page fully loaded in order to proceed with steps

File: cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js (hunk 2 above)
Comment (Alena-deriv) [reply cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js]: On mobile view datapicker is not working as expected; will be added in a separate PR (subtask https://app.clickup.com/t/20696747/CSIT-1120)

File: cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    ).should('be.visible')
+    cy.findByTestId('dt_mt5_password').type(
+      Cypress.env('credentials').test.diel.PSWD
+    )
+    cy.findByRole('button', { name: 'Create Deriv MT5 password' }).click()
+    cy.get('[class="dc-dialog__dialog"]').should(
+      'contain.text',
+      'Something’s not right'
+    )
+    //cy.c_visitResponsive('/reports/statement', 'large')
```
Comment (markodonnell-deriv) [inline cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js:52]: Could you remove the comments?

**Review by markodonnell-deriv — CHANGES_REQUESTED** [review]

File: cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js (hunk 3 above)
Comment (Alena-deriv) [reply cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js]: This is updated as requested

**Review by markodonnell-deriv — APPROVED** [review]

**Review by markodonnell-deriv — CHANGES_REQUESTED** [review]
Looks good

**Review by markodonnell-deriv — APPROVED** [review]
Looks good

File: cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import '@testing-library/cypress/add-commands'
+const addDays = (n) => {
+  let myDate = new Date()
+  myDate.setDate(myDate.getDate() + n)
+  return myDate.toISOString().slice(0, 10)
+}
+
+describe('QATEST-116798 Self Exclusion Session and login limits on desktop', () => {
+  beforeEach(() => {
+    cy.log('<E2EOAuthUrl - beforeEach>' + Cypress.env('oAuthUrl'))
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js:10]: No need to log this. Please remove.

File: cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    cy.findByRole('button', { name: 'Next' }).click()
+    cy.findByRole('button', { name: 'Accept' }).click()
+    cy.findByRole('button', { name: 'Yes, log me out immediately' }).click()
+    cy.c_login()
+    cy.c_visitResponsive('/appstore/traders-hub', 'large')
+    cy.findByTestId('dt_dropdown_display', { timeout: 10000 }).should('exist')
+    cy.findByTestId('dt_dropdown_display').click()
+    cy.get('#real', { timeout: 10000 }).should('exist')
+    cy.get('#real').click()
+    cy.get('[class="dc-text balance-text__text--real"]', {
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js:29]: Please use `findByDataTestId('dt_balance_text_container')`. You may use `cy.findAllByTestId('dt_balance_text_container').should('have.length', '2')` to verify page has loaded.

File: cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    cy.findByRole('button', { name: 'Next' }).click()
+    cy.findByText('Create a Deriv MT5 password').should('be.visible')
+    cy.findByText(
+      'You can use this password for all your Deriv MT5 accounts.'
+    ).should('be.visible')
+    cy.findByTestId('dt_mt5_password').type(
+      Cypress.env('credentials').test.diel.PSWD
+    )
+    cy.findByRole('button', { name: 'Create Deriv MT5 password' }).click()
+    cy.get('[class="dc-dialog__dialog"]').should(
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js:48]: Please try to use a testing library locator here. If not available, please change it to `cy.get('.dc-dialog__dialog')`

File: cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+      '/?chart_type=area&interval=1t&symbol=1HZ100V&trade_type=multiplier',
+      'large'
+    )
+    cy.findByRole(
+      'button',
+      { name: 'Up 10.00 USD' },
+      { timeout: 30000 }
+    ).should('exist')
+    cy.findByRole('button', { name: 'Up 10.00 USD' }).click()
+    cy.get('[class="dc-modal-body"]').should('be.visible') // checks Trade is unavailable
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js:64]: Please try to use a testing library locator here. If not available, please change it to `cy.get('.dc-modal-body')`

**Review by abuzar-deriv — CHANGES_REQUESTED** [review]
Requesting some changes.

File: cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js (hunk 2 above)
Comment (abuzar-deriv) [reply cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js]: What's the issue @Alena-deriv . You can type it out on responsive `cy.get(`input[type='Date']`).first().type(<Your date here>)`

File: cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js (hunk 4 above)
Comment (Alena-deriv) [reply cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js]: Removed

File: cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js (hunk 5 above)
Comment (Alena-deriv) [reply cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js]: This is changed as suggested

File: cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js (hunk 6 above)
Comment (Alena-deriv) [reply cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js]: Changed to `cy.get('.dc-dialog__dialog')` as was not able to use testing library locator here

File: cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js (hunk 7 above)
Comment (Alena-deriv) [reply cypress/e2e/wip/CSIT/selfExclusionSessionAndLoginLimits.cy.js]: Changed to `cy.get('.dc-modal-body')` as was not able to use testing library locator here

**Review by meeyun-deriv — APPROVED** [review]
LGTM

**Review by abuzar-deriv — APPROVED** [review]
LGTM

**Pull request merged**